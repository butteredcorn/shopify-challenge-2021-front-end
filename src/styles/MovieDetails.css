.movie-details-container {
    width: 90%;
    max-width: 1200px; 
}

.movie-details-container > * {
    margin: 1rem;
}

.main-section {
    display: flex;
    align-items: flex-start;
}

.main-section > * {
    margin: 1rem;
    width: 100%;
}