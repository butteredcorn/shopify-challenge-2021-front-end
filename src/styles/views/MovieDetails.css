.movie-details-container {
    width: 90%;
    max-width: 1200px; 
}

.movie-details-container > * {
    margin: 1rem;
}

.movie-details-main-section {
    display: flex;
    align-items: flex-start;
}

.movie-details-main-section > * {
    margin: 1rem;
    width: 100%;
}