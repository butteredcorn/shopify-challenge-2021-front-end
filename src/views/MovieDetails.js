import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom'
import styled from 'styled-components'
import { useQuery } from "@apollo/client"
import { GET_MOVIE_BY_ID } from '../graphql/Queries'


import '../styles/MovieDetails.css'

import MainHeader from '../components/MainHeader'
import GetMoviesByID from '../components/GetMovieByID';

const MovieDetailsContainer = styled.section`
    background-color: ${props => props.theme.bg.primary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`
const MainSection = styled.section`
    background-color: ${props => props.theme.bg.primary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const MovieDetails = () => {
    const location = useLocation()
    const id = location.state.id
    const { error, loading, data } = useQuery(GET_MOVIE_BY_ID, {variables: {id: id}});    
    const [movie, setMovie] = useState({})
    const [nominations, setNominations] = useState([])

    const addNomination = (nomination)  => {
        if (nominations && nominations.filter(n => n.imdbID === nomination.imdbID).length > 0) return;
        const updatedNominations = [...nominations, nomination]
        setNominations(updatedNominations)
        localStorage.setItem("nominations", JSON.stringify(updatedNominations))
    }

    const removeNomination = (id) => {
        const updatedNominations = nominations.filter(n => n.imdbID !== id)
        setNominations(updatedNominations)
        localStorage.setItem("nominations", JSON.stringify(updatedNominations))
    }

    useEffect(() => {
        if (data && data.movieByID) setMovie(data.movieByID)
    }, [data])

    useEffect(() => {
        const nominations = JSON.parse(localStorage.getItem("nominations"))
        setNominations(nominations)
    }, [])

    // useEffect(() => {
    //     localStorage.setItem("nominations", JSON.stringify(nominations))
    // }, [nominations])

    

    return (
        <MovieDetailsContainer className="movie-details-container">
            <MainHeader/>
            <MainSection className="main-section">
                <GetMoviesByID error={error} loading={loading} movie={movie} nominate={addNomination} remove={removeNomination}/>
            </MainSection>
        </MovieDetailsContainer>
    );
  };
  export default MovieDetails;