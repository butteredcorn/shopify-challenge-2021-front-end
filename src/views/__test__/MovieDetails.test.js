import { render, screen, cleanup, fireEvent, getByText } from '@testing-library/react';
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import ApolloClientProvider from '../../providers/ApolloProvider'
import ThemesProvider from '../../providers/ThemeProvider'
import MovieDetails from '../MovieDetails';
import {lightTheme, darkTheme} from '../../styles/common/theme.css'

const rgb2hex = (rgb) => `#${rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/).slice(1).map(n => parseInt(n, 10).toString(16).padStart(2, '0')).join('')}`

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      pathname: "localhost:4000/movie",
      state: "shopify_challenge_2021"
    })
}));

it('renders the Light Theme backgorund color for movie-details-container', () => {
    render(<ApolloClientProvider><BrowserRouter><ThemesProvider><MovieDetails /></ThemesProvider></BrowserRouter></ApolloClientProvider>);
    const MyHomeRoots = document.getElementsByClassName("movie-details-container")
    const style = window.getComputedStyle(MyHomeRoots[0])
    expect(rgb2hex(style.backgroundColor)).toBe(lightTheme.bg.primary)
});

it('renders the Dark Theme background color for movie-details-container after Theme toggle is clicked.', () => {
    const {container} = render(<ApolloClientProvider><BrowserRouter><ThemesProvider><MovieDetails /></ThemesProvider></BrowserRouter></ApolloClientProvider>);
    fireEvent(
        getByText(container, 'light Theme'),
        new MouseEvent('click', {
            bubbles: true,
            cancelable: true
        })
    )
    const MyHomeRoots = document.getElementsByClassName("movie-details-container")
    const style = window.getComputedStyle(MyHomeRoots[0])
    expect(rgb2hex(style.backgroundColor)).toBe(darkTheme.bg.primary)
});

it('matches a snapshot', () => {
  const json = renderer.create(<ApolloClientProvider><BrowserRouter><ThemesProvider><MovieDetails /></ThemesProvider></BrowserRouter></ApolloClientProvider>).toJSON();
  expect(json).toMatchSnapshot();
});