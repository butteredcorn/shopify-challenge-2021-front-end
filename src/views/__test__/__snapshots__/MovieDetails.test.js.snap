// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<section
  className="sc-eCImvq doEOAo movie-details-container"
>
  <section
    className="sc-bdvvaa cuEABP"
  >
    <header
      className="home-header"
    >
      <h1>
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          The Shoppies
        </a>
      </h1>
      <p
        className="toggle"
        onClick={[Function]}
      >
        light
         Theme
      </p>
    </header>
  </section>
  <section
    className="sc-jRQAMF kRaUyW movie-details-main-section"
  >
    <div
      className="sc-dkPtyc iKoOtC movie-container"
    >
      <p>
        Loading ...
      </p>
    </div>
  </section>
</section>
`;
